project.py

```diff
@@ -42,6 +42,8 @@ def Create(self):
             return ProjectOM()
         elif self.project == 'omi':
             return ProjectOMI()
+        elif self.project == 'omikits':
+		    return ProjectOMIKITS()
```

Comment:
I'm not wild on your choice of names here. `omi` is really OMI with unit tests, while `omikits` is really just building omi without unit tests (but building OMI).

I would prefer to see one target, `omi`, that did unit tests AND built the kits. Building the kits "blind" is risky at best, particularly since there would be no direct testing against them.

That would mean: Revert this and make a change in `regress` to do one more configure/build pass (to build the kits).

Reply:
It wouldn't quite be "doubling the build time". OMI builds a number of times (3 or 4, I think), to test different configurations of OMI. Adding this would probably add 10 minutes or so to the common case, although HP-UX would no doubt take longer (the machine is really slow).

In general, I don't believe we can leverage any of the existing builds. Running unit tests often makes different use of build flags that we don't necessarily want for release. But @palladia could comment on this for certain, I haven't looked at it lately. Certainly, with our SSL releases, that would be tricky to run on a local machine (when older versions of SSL aren't actually "installed", as such).

FWIW, I did do this myself in the past, particularly when test machines were unstable. I don't think I even bothered pushing it to GitHub, I just kept it locally. In my case, I just modified the `omi` rules since, at the time, I never cared about the unit tests.

But from a perspective of what's being checked in, I'm not a fan of not running unit tests on our products.

Reply:
Hi Jeff, how are you? Thanks for your comments. 
@johnkord Thanks for your comments also.

Sometimes, for testing/debug purpose or some else, we need a common tool to generate omi private build set like official build set. That is why we want to add a new project omikits in pbuild.  The omikits project has a different server list compared to omi project.  For sure, we can modify existing source python code (target & configuartionQuals & modify .puibld configuration file) to generate private builds.  But for convenience, I want to kick off a private build set by a simple command.

Reply:
Server list is controlled by the `.pbuild` file, the target has nothing to do with that directly. The target only specifies HOW the software is built, the `.pbuild` file determines everything else (what machines to use, location of the build directory, etc).

As John said, you can have a local branch, or you can fork this and then do what you want. But from an "official" mainline thing, I don't think it's a good idea to routinely skip unit tests. That's just sketchy software engineering practice.

My suggestion would be to revert this (although you could check in sample `.pbuild` file if you want) and then just add a simple change to `regress` (in OMI). Then you get unit tests AND your kits just by building OMI.

Reply:
Thanks! I have forked one to myself. 
I am closing this push request.